Sofa/framework/Core/src/sofa/core/collision/IntersectorFactory.h

```diff
@@ -52,6 +54,9 @@ class IntersectorFactory
     typedef BaseIntersectorCreator<TIntersectionClass> Creator;
     typedef std::vector<Creator*> CreatorVector;
     CreatorVector creatorVector;
+    // keep track of already created TIntersectorClass instances for each template combination
+    // when the factory is destroyed, the refcount of the instances is zero and they are cleaned up correctly
+    std::unordered_map<std::type_index, std::shared_ptr<void>> intersectorCache;
```

Comment:
Why is it necessary to store the pointer associated to a key, and not simply a list of pointers?

Reply:
There should only ever be one instance of the same template combination. So this is basically a hacked version of a singleton. If the factory creates the same type again, the old one is removed.

Reply:
Because that Template is not known in the parent :(

https://github.com/sofa-framework/sofa/blob/926377e46494638bc3e402a8ed7f54986ca1b6c5/Sofa/framework/Core/src/sofa/core/collision/IntersectorFactory.h#L39-L48